tests/test_calculator.hh

```diff
@@ -637,8 +650,15 @@ namespace rascal {
       // Count all the keys in the sparse gradient structure where the gradient
       // is nonzero (i.e. where the key has an entry in the structure)
       for (auto neigh : center) {
+        // TODO(max,felix) there should be a way to write "if neigh is ghost"
+        if (static_cast<size_t>(neigh.get_atom_tag()) >=
+            this->structure_manager->get_size()) {
+          // Don't compute gradient contributions onto ghost atoms
+          continue;
+        }
+        auto swapped_ref{std::move(swap_pair_ref(neigh).front())};
```

Comment:
`std::move` is suspicious here. `swap_pair_ref(neigh)` returns a new vector, but .front() returns a constant reference, on which std::move does nothing. Maybe you can have a different function to get the first value, or use `std::move(swap_pair_ref(neigh)[0])` which can do something, or ditch the std::move together if `PairRefKey_t` is trivial enough.

Reply:
It might no longer be necessary with PairRefKey_t; I haven't tried.

Reply:
Okay, removing the `std::move` does actually cause a compiler error because it's looking for the (deleted) copy constructor. So what exactly do you mean when you say "std::move does nothing" on this `vector.front()` reference?

Reply:
Hang on, `std::vector` can return either a const or a non-const reference: https://en.cppreference.com/w/cpp/container/vector/front. So how does it decide which one to return? Does it pick the non-const version because `move` only takes a non-const reference?

Reply:
If both are available it should pick one depending on the `const`ness of the vector itself, so I guess it picks the non-const one here.

Looks like a `&&` reference can be created from a `&` reference, so my bad here.